pretyping/vnorm.ml

```diff
@@ -239,8 +239,9 @@ and nf_stk ?from:(from=0) env sigma c t stk  =
       let (mib,mip) = Inductive.lookup_mind_specif env ind in
       let nparams = mib.mind_nparams in
       let params,realargs = Util.Array.chop nparams allargs in
+      let nparamdecls = Context.Rel.length (Inductive.inductive_paramdecls (mib,u)) in
       let pT =
-	hnf_prod_applist env (type_of_ind env (ind,u)) (Array.to_list params) in
+	hnf_prod_applist_assum env nparamdecls (type_of_ind env (ind,u)) (Array.to_list params) in
```

Comment:
Is this going to change anything? If I understand correctly, it could unfold more local definitions, if they come after arguments. But the next line does a `whd_all` anyway.

Reply:
Yes, I think I agree, this does not change the final semantics. The reason to do it is by separation of concerns. For instance, if you eventually change `whd_all` by some other reduction, or remove it, the instantiation of the telescope of the inductive type will remain correct while if the unfolding of the let's of the telescope is partly done by the function to instantiate the telescope and partly done with a further `whd_all`, this would not be true anymore.

Reply:
Isn't this morally replacing call by value by call by name on the `LetIn` part? I'm very tired so I may be completely wrong, but it seems that the new implementation will simply substitute the body of the `LetIn` before reducing it.